miden-lib/asm/miden/tx.masm

```diff
@@ -160,3 +160,35 @@ export.build_recipient_hash
   swapw hmerge
   # [RECIPIENT]
 end
+
+#! Executes the provided procedure against the foreign account. 
+#!
+#! Inputs:  [foreign_account_id, FOREIGN_PROC_ROOT, ...]
+#! Outputs: [<values returned from the foreign procedure>]
+export.execute_foreign_procedure
+    # pad the stack before the syscall
+    push.0.0 movup.2
+    exec.kernel_proc_offsets::start_foreign_context_offset
+    padw swapw padw padw swapdw
+    # => [offset, foreign_account_id, pad(14), FOREIGN_PROC_ROOT, ...]
+
+    # load the foreign account to the memory
+    syscall.exec_kernel_proc
+    # => [pad(16), FOREIGN_PROC_ROOT, ...]
+
+    # prepare the stack for the dyncall
+    dropw movupw.3
+    # => [FOREIGN_PROC_ROOT, pad(12), ...]
+
+    dyncall
+    # => [<values returned from the foreign procedure>]
+
+    # reset the current account data offset to the native offset (2048)
+    # WARNING: to be able to return the output of the foreign procedure its length should not exceed
+    # 12 elements.
```

Comment:
Should this warning be in the doc comments to the function? Basically, we cannot invoke a foreign procedure with more than 12 inputs and we cannot get more than 12 outputs back, right?

Reply:
Yes, right. I'll move this warning to the doc comments.